packages/lib/modules/tokens/token.helpers.ts

```diff
@@ -136,14 +136,24 @@ export function getLeafTokens(poolTokens: PoolToken[]) {
   return leafTokens
 }
 
-function getTokenOrUnderlying(token: PoolToken): TokenCore {
+function getTokenOrUnderlying(token: PoolToken): ApiToken {
   return token.isErc4626 && token.underlyingToken
     ? {
         ...token.underlyingToken,
         address: token.underlyingToken.address as Address,
-        index: token.index,
+        chainId: token.underlyingToken.chainId as number,
+        chain: token.underlyingToken.chain as GqlChain,
+        priority: token.underlyingToken.priority as number,
+        tradable: token.underlyingToken.tradable as boolean,
```

Comment:
Create a `buildApiToken` for doing this?

Reply:
Extracted `buildApiToken` and added a TODO that I will address with you in slack.